plugins/connection/incus.py

```diff
@@ -142,10 +142,16 @@ def _build_command(self, cmd) -> list[str]:
         ]
 
         if getattr(self._shell, "_IS_WINDOWS", False):
-            if (
-                (regex_match := self.powershell_regex_pattern.match(cmd))
-                and (regex_pattern := self.powershell_regex_pattern)
-            ) or ((regex_match := self.cmd_regex_pattern.match(cmd)) and (regex_pattern := self.cmd_regex_pattern)):
+            if regex_match_ps1 := self.powershell_regex_pattern.match(cmd):
+                regex_match = regex_match_ps1
+                regex_pattern = self.powershell_regex_pattern
+            elif regex_match_cmd := self.cmd_regex_pattern.match(cmd):
+                regex_match = regex_match_cmd
```

Comment:
Actually, we can take it a step further:
```suggestion
            if regex_match := self.powershell_regex_pattern.match(cmd):
                regex_pattern = self.powershell_regex_pattern
            elif regex_match := self.cmd_regex_pattern.match(cmd):
```